File: Python3/965.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def isUnivalTree(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        values = every(root)
        first = next(values)
        for v in values:
            if first != v:
                return False
        return True
        
def every(root):
    stack = [root]
    while stack:
        n = stack.pop()
        if n:
            yield n.val
            stack.append(n.left)
            stack.append(n.right)

File: Go/965.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isUnivalTree(root *TreeNode) bool {
    if root == nil {
        return true
    }
    
    this := true
    lv := root.Val
    rv := root.Val
    if root.Left != nil {
        lv = root.Left.Val
    }
    if root.Right != nil {
        rv = root.Right.Val
    }
    
    this = (root.Val == lv) && (root.Val == rv)
    return isUnivalTree(root.Left) && isUnivalTree(root.Right) && this
}
